src/main/java/training/supportbank/Transaction.java

```diff
@@ -0,0 +1,53 @@
+package training.supportbank;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+
+public class Transaction {
+
+    private final LocalDate date;
+    private final String from;
+    private final String to;
+    private final String narrative;
+    private final float amount;
+
+    public Transaction(String date, String from, String to, String narrative, String amount) {
+        this.date = getDate(date);
+        this.from = from;
+        this.to = to;
+        this.narrative = narrative;
+        this.amount = getAmount(amount);
+    }
+
+    public String getFrom() {return from;}
+
+    public String getTo() {return to;}
+
+    public float getAmount() {return amount;}
+
+    public LocalDate getDate() {
+        return date;
+    }
+
+    public String getNarrative() {
+        return narrative;
+    }
+
+    private LocalDate getDate(String date) {

```

Comment:
I think the method names are a little confusing here because the two `getDate` methods have different purposes - one lets other objects get the date from this object, and the other does some internal initialization. How could you make the difference clearer?

The same goes for `getAmount`.